opal/mca/accelerator/cuda/accelerator_cuda.c

```diff
@@ -217,12 +217,93 @@ static int accelerator_cuda_check_vmm(CUdeviceptr dbuf, CUmemorytype *mem_type,
     return 0;
 }
 
+static int accelerator_cuda_check_mpool(CUdeviceptr dbuf, CUmemorytype *mem_type,
+                                        int *dev_id)
+{
+#if OPAL_CUDA_VMM_SUPPORT
+    static int device_count = -1;
+    static int mpool_supported = -1;
+    CUresult result;
+    CUmemoryPool mpool;
+    CUmemAccess_flags flags;
+    CUmemLocation location;
+
+    if (device_count == -1) {
```

Comment:
The logic here is a little too lax for the cases with 0 devices. If I understand the code correctly, without a device (aka. without a valid device 0) the call to `cuDeviceGetAttribute` will never change `mpool_supported`, so the check `cuDeviceGetAttribute` will be executed every time. You should bail out of this function if `device_count` is 0.